### ISSUE TITLE
Is there a way to warn when someone has redundantly called `yield p = np`?

### ISSUE THREAD

**Issue by andrewculver** [bullet-train-co/nice_partials#34]
Going forward, because of https://github.com/bullet-train-co/nice_partials/pull/32, folks don't need to call `<% yield p = np %>`. If they do, it's not probably harmful except for performance... but is there some way of issuing a warning to them?

**Comment by kaspth** [comment]
There might be a way, but I've also been working on an update script to just do it for people.

This removes `<%= yield p = np %>\n+` and replaces `p.content_for` with `partial.content_for`, so it relies on #36 / #37.

```ruby
Dir["app/views/**/*.html.erb"].each do |path|
  if contents = File.read(path).match(/(<%=? yield\(?.*? = np\)? %>\n+)/m)&.post_match
    contents.gsub! /\bp\.(?=yield|helpers|content_for|content_for\?)/, "partial."
    File.write path, contents
  end
end
```

We could put this in the CHANGELOG to help people upgrade.

<details>
  <summary>Generated patch for `bullet_train-themes-light`</summary>

Note how this leaves `cp.content_for` unaltered in `_page.html.erb`.

```patch
diff --git a/app/views/themes/light/_box.html.erb b/app/views/themes/light/_box.html.erb
index 2556e5d..ecc95e8 100644
--- a/app/views/themes/light/_box.html.erb
+++ b/app/views/themes/light/_box.html.erb
@@ -1,48 +1,46 @@
-<% yield p = np %>
-
 <% divider ||= nil %>
 <% no_background ||= false %>
 <% title_size ||= nil %>
 <% title_padding ||= nil %>
 <% border_top ||= false %>
-<% body = p.content_for(:body) || p.content_for(:raw_body) %>
+<% body = partial.content_for(:body) || partial.content_for(:raw_body) %>
 <% pagy ||= nil %>
 
 <div class="<%= "bg-white rounded-md shadow dark:bg-darkPrimary-700" unless no_background %> overflow-hidden <%= border_top ? "border-t dark:border-darkPrimary-500" : "" %>">
   <div class="py-6 px-8 <%= title_padding %> space-y-2 <%= 'border-b shadow-sm dark:border-darkPrimary-600' if divider %>">
-    <% if p.content_for? :title %>
+    <% if partial.content_for? :title %>
       <h2 class="text-xl font-semibold dark:text-white <%= title_size %>">
-        <%= p.content_for :title %>
+        <%= partial.content_for :title %>
       </h2>
     <% end %>
 
-    <% if p.content_for? :description %>
+    <% if partial.content_for? :description %>
       <p class="text-gray-400 font-light leading-normal">
-        <%= p.content_for :description %>
+        <%= partial.content_for :description %>
       </p>
     <% end %>
   </div>
 
   <div class="space-y-4">
-    <% if p.content_for? :table %>
+    <% if partial.content_for? :table %>
       <div class="box-table <%= divider ? 'mt-4' : '-mt-1' %> pb-0.5">
-        <%= p.content_for :table %>
+        <%= partial.content_for :table %>
       </div>
     <% end %>
 
     <% if body %>
-      <div class="<%= "pt-7 px-8 space-y-7 #{p.content_for?(:actions) ? 'pb-3' : 'pb-7'}" unless p.content_for?(:raw_body) %>">
+      <div class="<%= "pt-7 px-8 space-y-7 #{partial.content_for?(:actions) ? 'pb-3' : 'pb-7'}" unless partial.content_for?(:raw_body) %>">
         <div class="space-y-4 <%= '-mt-4' unless divider %>">
           <%= body %>
         </div>
```
</details>

**Comment by kaspth** [comment]
In #45 I'm making it such that we'll avoid auto-yielding if there's a manual yield in the partial. So that should make upgrades easier, and then people could use the script above.

**Comment by kaspth** [comment]
Solved in #45 where we only auto-yield if there's no manual yield, and the update script is in the CHANGELOG for people to help themselves. I think that's enough coverage to skip needing a warning.
